### ISSUE TITLE
'Undefined index: strPrivDescDeleteHistoricalRows' in 'User account'

### ISSUE THREAD

**Issue by aschuch247** [phpmyadmin/phpmyadmin#15308]
### Describe the bug

Undefined index: strPrivDescDeleteHistoricalRows

```
Notice in ./libraries/classes/Server/Privileges.php#435
Undefined index: strPrivDescDeleteHistoricalRows

Backtrace

./libraries/classes/Server/Privileges.php#180: PhpMyAdmin\Server\Privileges::getGrantsArray()
./libraries/classes/Server/Privileges.php#3540: PhpMyAdmin\Server\Privileges::extractPrivInfo(
array,
boolean true,
)
./libraries/classes/Server/Privileges.php#4744: PhpMyAdmin\Server\Privileges::getUsersOverview(
,
array,
string './themes/pmahomme/img/',
string 'ltr',
)
server_privileges.php#438: PhpMyAdmin\Server\Privileges::getHtmlForUserOverview(
string './themes/pmahomme/img/',
string 'ltr',
)
```

```
Notice in ./libraries/classes/Server/Privileges.php#440
Undefined index: strPrivDescDeleteHistoricalRows

Backtrace

./libraries/classes/Server/Privileges.php#180: PhpMyAdmin\Server\Privileges::getGrantsArray()
./libraries/classes/Server/Privileges.php#3540: PhpMyAdmin\Server\Privileges::extractPrivInfo(
array,
boolean true,
)
./libraries/classes/Server/Privileges.php#4744: PhpMyAdmin\Server\Privileges::getUsersOverview(
,
array,
string './themes/pmahomme/img/',
string 'ltr',
)
server_privileges.php#438: PhpMyAdmin\Server\Privileges::getHtmlForUserOverview(
string './themes/pmahomme/img/',
string 'ltr',
)
```

### To Reproduce

Steps to reproduce the behaviour:

* Log in to phpMyAdmin 4.9.0.1 as administrative user and open 'User account' tab.
* Notice the notices.

### Expected behaviour

Do not see the notices. (Actually, do not see them because they do not happen.)

### Server configuration

- phpMyAdmin version: 4.9.0.1
- `Server version: 5.6.44-86.0-log - Percona Server (GPL), Release 86.0, Revision eba1b3f`

### Additional context

phpMyAdmin 4.8.5 and 4.9.0.1 are installed in parallel (different directories). There is no such error using phpMyAdmin 4.8.5 on the same server.

**Comment by williamdes** [comment]
Duplicate of #15229

**Comment by williamdes** [comment]
https://github.com/phpmyadmin/phpmyadmin/blob/9abd308837bb7c3424c6aaa01e1ce180db83413a/server_privileges.php#L90

**Comment by williamdes** [comment]
91c758a

**Comment by williamdes** [comment]
@aschuch247 Where did you have this php notice ?
I had it and fixed it on column level permissions interface.

**Comment by williamdes** [comment]
@aschuch247 Please when you see the notices on the phpMyAdmin interface, report them using the "Report" button.

**Comment by aschuch247** [comment]
91c758aae4ec9f10fb32b0d28c395211d4654e0f is included in 4.9.0.1 already? It seems that way. But I still have this issue in phpMyAdmin 4.9.0.1.

@williamdes: As reported, I have this directly in the `User accounts` overview below the table labelled as `User accounts overview`.

The server is inside a VPN, so I just followed instructions: `An error has been detected and an error report has been generated but failed to be sent. If you experience any problems please submit a bug report manually. You may want to refresh the page.`

**Comment by williamdes** [comment]
> 91c758a is included in 4.9.0.1 already? It seems that way. But I still have this issue in phpMyAdmin 4.9.0.1.

Yes, it is included :)

@aschuch247 Can you send more informations on what database server you are using ?
I think you are using MariaDB

**Comment by aschuch247** [comment]
@williamdes: No, I am using Percona: `Server version: 5.6.44-86.0-log - Percona Server (GPL), Release 86.0, Revision eba1b3f`.

**Comment by williamdes** [comment]
@aschuch247 I used the same version you have and could not reproduce the behavior
can you please send `SHOW GRANTS;` ?

**Comment by aschuch247** [comment]
@williamdes: https://dev.mysql.com/doc/refman/5.6/en/proxy-users.html, see section `Default Proxy Users`.

```sql
GRANT ALL PRIVILEGES ON *.* TO 'root'@'localhost' IDENTIFIED BY PASSWORD '*secret' WITH GRANT OPTION
GRANT PROXY ON ''@'' TO 'root'@'localhost' WITH GRANT OPTION
```

**Comment by williamdes** [comment]
@aschuch247 Can you please try to see if you have the line https://github.com/phpmyadmin/phpmyadmin/blob/9abd308837bb7c3424c6aaa01e1ce180db83413a/server_privileges.php#L90

I can not reproduce this issue and spent hours trying to..
I have a fix but it is not very cool.


Can you send a step by step way to reproduce this issue ?
And a screenshot ?

**Comment by williamdes** [comment]
```patch
diff --git a/libraries/classes/Server/Privileges.php b/libraries/classes/Server/Privileges.php
index 75823fcbc2..bd07df4db4 100644
--- a/libraries/classes/Server/Privileges.php
+++ b/libraries/classes/Server/Privileges.php
@@ -432,12 +432,12 @@ class Privileges
             array(
                 'Delete_history_priv',
                 'DELETE HISTORY',
-                $GLOBALS['strPrivDescDeleteHistoricalRows']
+                __('Allows deleting historical rows.')
             ),
             array(
                 'Delete versioning rows_priv',
                 'DELETE HISTORY',
-                $GLOBALS['strPrivDescDeleteHistoricalRows']
+                __('Allows deleting historical rows.')
             ),
             array(
                 'Create_routine_priv',
```

**Comment by aschuch247** [comment]
@williamdes: I debugged the issue myself now, but all I tried did not yield any result. Long story short, it now works after restarting PHP FPM.

I never looked into how PHP FPM works and how it does stuff (like (OP code) caching), but there seems to be a bug somewhere in there. So if a symlink gets changed from one phpMyAdmin version to another, this is not (fully) noticed by PHP FPM. The version of phpMyAdmin worked well enough, so that this was not noticed before.

I feel sorry for wasting your time and consider my time investment as lesson learned to pay more attention when updating phpMyAdmin.
